backend/cmd/headlamp.go

```diff
@@ -142,6 +148,73 @@ func StartHeadlampServer(config *HeadlampConfig) {
 	// Configuration
 	r.HandleFunc("/config", clientConf.getConfig).Methods("GET")
 
+	state := "app"
+	ctx := context.Background()
+	oauthRequestQueue := make([]*OauthConfig, 0)
+
+	r.HandleFunc("/oidc", func(w http.ResponseWriter, r *http.Request) {
+		cluster := r.URL.Query().Get("cluster")
+
+		oauthConfig, err := GetClusterOauth2Config(cluster)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		oauthRequestQueue = append(oauthRequestQueue, oauthConfig)
+		http.Redirect(w, r, oauthConfig.Config.AuthCodeURL(state), http.StatusFound)
+	}).Queries("cluster", "{cluster}")
+
+	r.HandleFunc("/oidc-callback", func(w http.ResponseWriter, r *http.Request) {
+		if len(oauthRequestQueue) == 0 {
+			http.Error(w, "no request found for this callback", http.StatusBadRequest)
+			return
+		}
+		var oauthConfig *OauthConfig
+		oauthConfig, oauthRequestQueue = oauthRequestQueue[0], oauthRequestQueue[1:]
```

Comment:
So I imagine the match by URL is not possible here? (i.e. `r.URL` to match the clusters, as we had talked about)